packages/components/tests/dummy/app/templates/components/dropdown.hbs

```diff
@@ -79,424 +130,550 @@
-    component:
-  </p>
-  <dl class="dummy-component-props" aria-labelledby="listitem-api">
-    <dt>
-      item
-      <code>
-        enum
-      </code>
-    </dt>
+
+  <h4 class="dummy-h4">ListItem::CopyItem</h4>
+  {{! TODO: add props here if we decide to use the current version }}
+  <p class="dummy-paragraph">TODO??</p>
+
+  <h4 class="dummy-h4">ListItem::Description</h4>
+  <p class="dummy-paragraph">Here is the API for the "description" list item component (yielded as
+    <code class="dummy-code">Description</code>):</p>
+  <dl class="dummy-component-props" aria-labelledby="listitem-description-api">
+    <dt>text <code>string</code></dt>
     <dd>
-      <p>
-        Acceptable values:
-      </p>
-      <ol>
-        <li class="default">interactive</li>
-        <li>copy-item</li>
-        <li>description</li>
-        <li>generic</li>
-        <li>separator</li>
-        <li>title</li>
-      </ol>
-      <p>🚨 Important Note: the
-        <span class="dummy-code">generic</span>
-        is a yield and the developer is completely responsible for any item yielded there (including the accessibility
-        of that item). This is provided as an escape hatch to empower developers to more easily transition to the design
-        system. It should be considered as a temporary measure with plans for replacement.</p>
+      <p>The text to be used for the description.</p>
+      <p><em>If no text value is defined an error will be thrown.</em></p>
+    </dd>
+    <dt>...attributes</dt>
+    <dd>
+      <p><code class="dummy-code">...attributes</code> spreading is supported on this component.</p>
+    </dd>
+  </dl>
+
+  <h4 class="dummy-h4">ListItem::Generic</h4>
+  <p class="dummy-paragraph">Here is the API for the "generic" list item component (yielded as
+    <code class="dummy-code">Generic</code>):</p>
+  <dl class="dummy-component-props" aria-labelledby="listitem-generic-api">
+    <dt>"yield"</dt>
+    <dd>
+      <p>Elements passed as children of this sub-component are yielded inside the list item.</p>
+      <p>🚨
+        <em>Notice: when using the "generic" list item the developer is completely responsible for any item yielded
+          there (including the accessibility of that item).</em></p>
     </dd>
-    <dt>text
-      <code>string</code>
-    </dt>
+    <dt>...attributes</dt>
     <dd>
-      <p>The text to be used for all items. The only items that do not require text to be defined are the separator and
-        generic item types. Otherwise,
-        <em>If no text value is defined an error will be thrown.</em></p>
+      <p><code class="dummy-code">...attributes</code> spreading is supported on this component.</p>
+    </dd>
+  </dl>
+
+  <h4 class="dummy-h4">ListItem::Interactive</h4>
+  <p class="dummy-paragraph">Here is the API for the "interactive" list item component (yielded as
+    <code class="dummy-code">Interactive</code>):</p>
+  <dl class="dummy-component-props" aria-labelledby="listitem-generic-api">
```

Comment:
Looking through this PR, I see these lines repeated, although we don't do this in other parts of the documentation. 

We do need something that provides the label for the `<dl>` element (notice the `aria-labelledby` attribute on the `<dl>`, whose value is an `id` attribute value from another element) - but perhaps we could put that on the `<h4>` for the section. WDYT?

Reply:
I have done all the associations in https://github.com/hashicorp/design-system/pull/235/commits/403789977cba3d2575a624b790df113b122014ea